.github/workflows/docker.yml

```diff
@@ -0,0 +1,37 @@
+name: Docker Image CI
+
+on:
+  workflow_dispatch:
+  release:
+    types: [created]
+    
+jobs:
+  build-and-push:
+    runs-on: github-hosted-heavy-runner
+    steps:
+
+    - 
+      name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+      
+    - 
+      name: Log in to Docker registry
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    -
+      name: Set RELEASE_TAG env
+      run: echo "RELEASE_TAG=${{ github.event.release.tag_name || 'latest' }}" >> $GITHUB_ENV
+
+    - name: Build and push Docker image
+      uses: docker/build-push-action@v5
+      with:
+        file: Dockerfile
+        push: true
+        tags: |
```

Comment:
From looking at the code I am not sure about one thing. Do we handle `latest` in the correct way?

For example if I release `rc.2` version I don't want it to be `latest`. 


1. Release 0.28.0-1.37.3 (new latest)
2. Release 0.28.0-1.38.0-rc.2 (fresh release, but not new latest, new experimental, should not become docker latest)
3. Release 0.28.0-1.38.0 (fresh release and new latest, because stable)

Reply:
If we can agree, that pattern for version that is ready to become latest is x.x.x-x.x.x, then we can check that pattern and only then tag with latest. Otherwise just build and push only specific release. WDYT?

Reply:
Yep, sounds good to me.